frontend/utils/giteaApi.js

```diff
@@ -0,0 +1,229 @@
+import path from 'path'
+import slugify from 'slugify'
+
+const giteaApiUrl = `${process.env.KITSPACE_GITEA_URL}/api/v1`
+const credentials = 'include'
+const mode = 'cors'
+const headers = { 'Content-Type': 'application/json' }
+
+/**
+ * Create a new Gitea repo
+ * @param name {string}
+ * @param description {string}
+ * @param csrf {string}
+ * @returns {Promise<string>}
+ */
+export const createRepo = async (name, description, csrf) => {
+  const endpoint = `${giteaApiUrl}/user/repos?_csrf=${csrf}`
+  const giteaOptions = {
+    _csrf: csrf,
+    name: slugify(name),
+    description: description,
+    repo_template: '',
+    issue_labels: '',
+    gitignores: '',
+    license: 'MIT',
+    readme: 'Default',
+    auto_init: true,
+    private: false,
+    default_branch: 'master',
+  }
+  const res = await fetch(endpoint, {
+    method: 'POST',
+    credentials,
+    mode,
+    headers,
+    body: JSON.stringify(giteaOptions),
+  })
+
+  const body = await res.json()
+
+  return res.ok ? body['full_name'] : ''
+}
+
+/**
+ * Update a repo with new fields
+ * @param repo {string}
+ * @param updateFields{Object}
+ * @param csrf{string}
+ * @returns {Promise<boolean>}
+ */
+export const updateRepo = async (repo, updateFields, csrf) => {
+  const endpoint = `${giteaApiUrl}/repos/${repo}?_csrf=${csrf}`
+
+  const res = await fetch(endpoint, {
+    method: 'PATCH',
+    credentials,
+    mode,
+    headers,
+    body: JSON.stringify(updateFields),
+  })
+
+  return res.ok
+}
+
+/**
+ * Mirror an existing remote git repo to a Gitea repo
+ * @param remoteRepo {string} url of the remote repo
+ * @param uid {string}
+ * @param csrf {string}
+ * @returns {Promise<boolean>}
+ */
+export const migrateRepo = async (remoteRepo, uid, csrf) => {
+  const clone_addr = remoteRepo
+  const repo_name = urlToName(clone_addr)
+  const endpoint = `${giteaApiUrl}/repos/migrate?_csrf=${csrf}`
+  const giteaOptions = {
+    clone_addr,
+    uid,
+    repo_name,
+    mirror: false,
+    wiki: false,
+    private: false,
+    pull_requests: false,
+    releases: true,
+  }
+
+  const res = await fetch(endpoint, {
+    method: 'POST',
+    mode,
+    credentials: 'include',
+    headers,
+    body: JSON.stringify(giteaOptions),
+  })
+
+  return res.ok
+}
+/**
+ * delete the corresponding gitea repo for a project.
+ * @param repo {string}
+ * @param csrf {string}
+ * @returns {Promise<boolean>}
+ */
+export const deleteRepo = async (repo, csrf) => {
+  const endpoint = `${giteaApiUrl}/repos/${repo}?_csrf=${csrf}`
+  const res = await fetch(endpoint, {
+    method: 'DELETE',
+    mode,
+    credentials,
+    headers,
+  })
+
+  return res.ok
+}
+
+/**
+ * Get list of files in a gitea repo.
+ * @param repo {string}
+ * @param csrf {string}
+ * @returns {Promise<Array|null>}
+ */
+export const getRepoFiles = async (repo, csrf) => {
+  const endpoint = `${giteaApiUrl}/repos/${repo}/contents?ref=master&_csrf=${csrf}`
```

Comment:
Just leaving a note here: we can't assume the default branch will be `master`.

Reply:
This repo is created when a user uploads files through drag and drop **not** through `sync`. So I guess we are free to upload to whichever branch we want. We shouldn't guess the branch from which the assets will be generated but this will be a separate feature.

Reply:
So `getRepoFiles` won't work on repos that are synced?

Reply:
Oh, sorry I thought the commit was on a createRepo. You're right.

Reply:
- Added optional param for the branch in 9513187.
- Though haven't updated the code that depends on `getRepoFiles` to fetch from the main branch yet.
- We might take the same approach as GitHub pages which builds the website from the main branch and ignores the rest.

p.s: I tried `HEAD`, it didn't work.